lib/util.js

```diff
@@ -804,3 +805,88 @@ exports.validatePath = function (dir) {
     return false;
   }
 };
+
+exports.getEnvVar = function (variable) {
+  return process.env[variable.toLowerCase()] || process.env[variable.toUpperCase()];
+};
+
+exports.validateProxy = function (proxy) {
+  const { host, port, noProxy, user, password } = proxy;
+  // check for missing proxyHost
+  Errors.checkArgumentExists(this.exists(host),
+    ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_HOST);
+
+  // check for invalid proxyHost
+  Errors.checkArgumentValid(this.isString(host),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_HOST);
+
+  // check for missing proxyPort
+  Errors.checkArgumentExists(this.exists(port),
+    ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PORT);
+
+  // check for invalid proxyPort
+  Errors.checkArgumentValid(this.isNumber(port),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PORT);
+
+  if (this.exists(noProxy)) {
+    // check for invalid noProxy
+    Errors.checkArgumentValid(this.isString(noProxy),
+      ErrorCodes.ERR_CONN_CREATE_INVALID_NO_PROXY);
+  }
+
+  if (this.exists(user) || this.exists(password)) {
+    // check for missing proxyUser
+    Errors.checkArgumentExists(this.exists(user),
+      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_USER);
+
+    // check for invalid proxyUser
+    Errors.checkArgumentValid(this.isString(user),
+      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_USER);
+
+    // check for missing proxyPassword
+    Errors.checkArgumentExists(this.exists(password),
+      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PASS);
+
+    // check for invalid proxyPassword
+    Errors.checkArgumentValid(this.isString(password),
+      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PASS);
+
+  } else {
+    delete proxy.user;
+    delete proxy.password;
+  }
+};
+
+exports.validateEmptyString = function (value) {
+  return value !== '' ? value : undefined;
+};
+
+exports.getProxyEnv = function (isHttps) {
+  let proxyFromEnv = isHttps ? this.getEnvVar('https_proxy') : this.getEnvVar('http_proxy');
+  const protocol = isHttps ? 'https' : 'http';
```

Comment:
Same as above. If users want to use HTTP://proxy in HTTPS_PROXY, they should set HTTPS_PROXY with the schema like http://. As far as I know, other drivers and Axios will be the same

Reply:
the difficult thing is with these environmental variables is that they are **environmental**. They affect the whole environment, all the applications running in the same environment, not just snowflake-sdk.

The main goal is for snowflake-sdk to play nice with the environmental variables, which already work well with any other application. 

I still think it would be good to consider my suggestion above, i.e. prepending http and logging a line, _if_ HTTPS_PROXY and/or HTTP_PROXY is configured, for some reason, as 'my.pro.xy:1234' instead of 'http://my.pro.xy:1234' (or https://my.pro.xy:1234)

Reply:
Hello @sfc-gh-dszmolka, it is my bad. I have researched this, and I realized that other drivers have used the 'HTTP' schema if there is no schema in the proxy URL ex) python(request), or curl. I updated my PR based on your comment.